### PR TITLE
Add a config `hash_in_kernel_names` 

### PR DESCRIPTION
**Pull request by FindHao**
This option helps correlating triton kernel in profiling traces with its source code. It is disabled by default and should be for profiling purpose only.

After run with `TORCHINDUCTOR_HASH_IN_KERNEL_NAMES=1`, an exampled code snippet in output_code.py:
```
# kernel path: /tmp/torchinductor_yhao/tmp7mf5xcsz/lo/clobi4dt6e3xuk6pckv5l7uqxsf2ianp6d3g6lgh6wrm5f64vxaz.py
# Topologically Sorted Source Nodes: [embedding], Original ATen: [aten.embedding]
# Source node to ATen node mapping:
#   embedding => embedding
# Graph fragment:
#   %embedding : [num_users=1] = call_function[target=torch.ops.aten.embedding.default](args = (%arg2_1, %view), kwargs = {})
triton_poi_fused_embedding_1_cemi5yor3uh3muxgzlaoqs5wrpjk7q6xod6dgmymw2y5xd6wk3sk = async_compile.triton('triton_poi_fused_embedding_1_cemi5yor3uh3muxgzlaoqs5wrpjk7q6xod6dgmymw2y5xd6wk3sk', 
```
The hash is per-kernel and different from the hashed file names in `/tmp/`.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Comment by adamomainz** [comment]
Thanks for putting this up! Can you post an example of the kernel names with the functionality enabled?

**Comment by adamomainz** [comment]
Spoke offline and saw the output. Looks great! will review once out of draft

**Comment by Chillee** [comment]
If we don't turn this on by default how can this be used fleet-wide?

**Comment by FindHao** [comment]
> If we don't turn this on by default how can this be used fleet-wide?

@briancoutinho does durin runs with configurable environments?

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -3157,6 +3157,8 @@ def define_kernel(self, src_code, node_schedule, kernel):
             kernel_name = "_".join(
                 ["triton", kernel_category, fused_name, wrapper.next_kernel_suffix()]
             )
+            if config.triton.hash_in_kernel_names:
+                kernel_name = f"{kernel_name}_{code_hash(src_code.strip())}"
```
Comment (briancoutinho) [inline torch/_inductor/codegen/triton.py:3161]: nit 
kernel_name += f"_{code_hash(src_code.strip())}"

**Comment by FindHao** [comment]
Close this PR since this issue is solved by another way. https://github.com/pytorch/pytorch/pull/139135

**Pull request closed without merging**